lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -299,50 +301,20 @@ class ThreadRepositoryImpl extends ThreadRepository {
   }
 
   @override
-  Future<bool> emptyTrashFolder(AccountId accountId, MailboxId trashMailboxId) async {
-    var finalResult = true;
-    var hasEmails = true;
-
-    while (hasEmails) {
-      Email? lastEmail;
-
-      final emailsResponse = await mapDataSource[DataSourceType.network]!.getAllEmail(
-          accountId,
-          sort: <Comparator>{}
-            ..add(EmailComparator(EmailComparatorProperty.receivedAt)
-              ..setIsAscending(false)),
-          filter: EmailFilterCondition(inMailbox: trashMailboxId, before: lastEmail?.receivedAt),
-          properties: Properties({EmailProperty.id}));
-
-      if (emailsResponse.state != null) {
-        await _updateState(emailsResponse.state!);
-      }
-
-      var newEmailList =  emailsResponse.emailList ?? <Email>[];
-      if (lastEmail != null) {
-        newEmailList = newEmailList.where((email) => email.id != lastEmail!.id).toList();
-      }
-
-      log('ThreadRepositoryImpl::emptyTrashFolder(): ${newEmailList.length}');
-
-      if (newEmailList.isNotEmpty == true) {
-        lastEmail = newEmailList.last;
-        hasEmails = true;
-        final emailIds = newEmailList.map((email) => email.id).toList();
-
-        final listEmailIdDeleted = await emailDataSource.deleteMultipleEmailsPermanently(accountId, emailIds);
+  Future<List<EmailId>> emptyTrashFolder(
+      AccountId accountId,
+      MailboxId trashMailboxId,
+  ) async {
 
-        if (listEmailIdDeleted.isNotEmpty && listEmailIdDeleted.length == emailIds.length) {
+    return mapDataSource[DataSourceType.network]!.emptyTrashFolder(
+        accountId,
+        trashMailboxId,
+        (state) async {
+          await _updateState(state);
+        },
+        (listEmailIdDeleted) async {
```

Comment:
is it ok here? Update cache without update state. I am so worried about it. @ManhNTX @dab246

Reply:
@hoangdat i updated state at line 313 please check it